diff --git a/umbrielfx/include/umbrielfx/types/fx/clipped_region.h b/umbrielfx/include/umbrielfx/types/fx/clipped_region.h
index 72b088ab1d..0f7c1ff5f8 100644
--- a/umbrielfx/include/umbrielfx/types/fx/clipped_region.h
+++ b/umbrielfx/include/umbrielfx/types/fx/clipped_region.h
@@ -15,7 +15,7 @@
 
 #define CORNER_RADIUS_MAX (UINT16_MAX)
 
-static __always_inline uint16_t corner_radius_clamp(int radius) {
+static inline __attribute__((always_inline)) uint16_t corner_radius_clamp(int radius) {
 	if (radius <= 0) {
 		return 0;
 	}
@@ -27,7 +27,7 @@
 	return radius;
 }
 
-static __always_inline struct fx_corner_radii corner_radii_new(int top_left, int top_right, int bottom_right, int bottom_left) {
+static inline __attribute__((always_inline)) struct fx_corner_radii corner_radii_new(int top_left, int top_right, int bottom_right, int bottom_left) {
 	return (struct fx_corner_radii) {
 		corner_radius_clamp(top_left),
 		corner_radius_clamp(top_right),
@@ -36,15 +36,15 @@
 	};
 }
 
-static __always_inline struct fx_corner_radii corner_radii_all(int radius) {
+static inline __attribute__((always_inline)) struct fx_corner_radii corner_radii_all(int radius) {
 	return corner_radii_new(radius, radius, radius, radius);
 }
 
-static __always_inline struct fx_corner_radii corner_radii_none(void) {
+static inline __attribute__((always_inline)) struct fx_corner_radii corner_radii_none(void) {
 	return corner_radii_all(0);
 }
 
-#define corner_radii_func(name, tl, tr, br, bl) static __always_inline struct fx_corner_radii corner_radii_##name(int radius) { return corner_radii_new(tl, tr, br, bl); }
+#define corner_radii_func(name, tl, tr, br, bl) static inline __attribute__((always_inline)) struct fx_corner_radii corner_radii_##name(int radius) { return corner_radii_new(tl, tr, br, bl); }
 
 corner_radii_func(top, radius, radius, 0, 0);
 corner_radii_func(bottom, 0, 0, radius, radius);
@@ -63,7 +63,7 @@
  *
  * This can be compared to an `AND` operation
  */
-static __always_inline struct fx_corner_radii fx_corner_radii_filter(struct fx_corner_radii input, struct fx_corner_radii filter) {
+static inline __attribute__((always_inline)) struct fx_corner_radii fx_corner_radii_filter(struct fx_corner_radii input, struct fx_corner_radii filter) {
 	return corner_radii_new(
 		input.top_left && filter.top_left ? input.top_left : 0,
 		input.top_right && filter.top_right ? input.top_right : 0,
@@ -77,7 +77,7 @@
  *
  * This can be compared to an `OR` operation
  */
-static __always_inline struct fx_corner_radii fx_corner_radii_pick(struct fx_corner_radii lhs, struct fx_corner_radii rhs) {
+static inline __attribute__((always_inline)) struct fx_corner_radii fx_corner_radii_pick(struct fx_corner_radii lhs, struct fx_corner_radii rhs) {
 	return corner_radii_new(
 		lhs.top_left ? lhs.top_left : rhs.top_left,
 		lhs.top_right ? lhs.top_right : rhs.top_right,
diff --git a/umbrielfx/internal/types/fx/clipped_region.h b/umbrielfx/internal/types/fx/clipped_region.h
index 6f3a4d434f..62f8b397f2 100644
--- a/umbrielfx/internal/types/fx/clipped_region.h
+++ b/umbrielfx/internal/types/fx/clipped_region.h
@@ -20,7 +20,7 @@
     struct fx_corner_fradii corners;
 };
 
-static __always_inline bool clipped_fregion_is_valid(const struct clipped_fregion *fregion) {
+static inline __attribute__((always_inline)) bool clipped_fregion_is_valid(const struct clipped_fregion *fregion) {
 	return !fx_corner_fradii_is_empty(&fregion->corners) && !wlr_box_empty(&fregion->area);
 };
 
